### ISSUE TITLE
Input element is not fully controlled

### ISSUE THREAD

**Issue by REX500** [grommet/grommet#5054]
Input element doesn't reset it's internal value when new value is changed from a string to null

### Expected Behavior

Input should show the same value that we pass to it.

### Actual Behavior

When input value set from a string to null, input keeps showing the old value... You would think that it's uncontrolled but it isn't!
Inputs should either be controlled or uncontrolled. Grommet is probably copying the value and doing some magic on it and then manually setting it for the input. This is dangerous behaviour and we can see that in this case leads to errors.

### URL, screen shot, or Codepen exhibiting the issue

Codesandbox example of the issue: https://codesandbox.io/s/grommet-input-438rz?file=/src/index.js

### Steps to Reproduce

Go to https://codesandbox.io/s/grommet-input-438rz?file=/src/index.js and follow the flow and you will quickly get it.

### Your Environment

MacOS, Chrome....

- Grommet version: newest at the time of testing -> 2.16.3
- Browser Name and version: Chrome -> 89.0.4389.72
- Operating System and version (desktop or mobile): MacOS desktop -> 11.01 Big Sur

**Comment by taysea** [comment]
@REX500 I don't think this is a Grommet bug but instead an implementation pattern that React would not recommend. I recreated the codesandbox you shared locally in storybook (thanks for such detailed steps, it was really helpful). When I opened the DevTools, I first noticed this warning that "`value` prop on input should not be null, change to empty string or undefined instead":
<img width="337" alt="Screen Shot 2021-03-24 at 4 37 35 PM" src="https://user-images.githubusercontent.com/12522275/112398043-86510c00-8cc0-11eb-96fd-53a889f26c50.png">

After seeing that, I try changing the places you have `null` to `undefined`. I also set up a parallel example with a non-Grommet input using the following code to see if this was Grommet specific or something React would discourage in general:
```
<input
     value={value}
     placeholder="See placeholder when there's no value..."
    onChange={e => setValue(e.target.value)}
/>
```
When I typed into both the Grommet input and into this non-Grommet input, in both cases I got the console warning "A component is changing an uncontrolled input to be controlled."
<img width="337" alt="Screen Shot 2021-03-24 at 4 44 39 PM" src="https://user-images.githubusercontent.com/12522275/112398184-d7f99680-8cc0-11eb-9633-e58af5b9b6f9.png">

If you set your value to `""` (empty string) instead of undefined or null, all works fine and it seems that is what React is recommending since the console warning appear when I'm using a normal `input` as well.
